dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/dxf2/sync/TrackerSynchronizationTest.java

```diff
@@ -31,180 +31,76 @@
-
-    @Autowired
-    private TrackedEntityInstanceAuditService trackedEntityInstanceAuditService;
-
-    @Autowired
-    private TrackedEntityTypeService trackedEntityTypeService;
-
-    @Autowired
-    private Notifier notifier;
-
-    @Autowired
-    private ObjectMapper jsonMapper;
-
-    @Autowired
-    private TrackedEntityInstanceAggregate trackedEntityInstanceAggregate;
-
-    @Autowired
-    private TrackedEntityAttributeStore trackedEntityAttributeStore;
-
-    @Autowired
-    @Qualifier( "xmlMapper" )
-    private ObjectMapper xmlMapper;
-
     private TrackedEntityInstanceService subject;
 
     private TrackedEntityInstanceQueryParams queryParams;
 
     private TrackedEntityInstanceParams params;
 
-    private Session currentSession;
-
     private void prepareDataForTest()
     {
         TrackedEntityAttribute teaA = createTrackedEntityAttribute( 'a' );
         TrackedEntityAttribute teaB = createTrackedEntityAttribute( 'b' );
         teaB.setSkipSynchronization( true );
-        currentSession.save( teaA );
-        currentSession.save( teaB );
+        manager.save( teaA );
+        manager.save( teaB );
         TrackedEntityType tet = createTrackedEntityType( 'a' );
         TrackedEntityTypeAttribute tetaA = new TrackedEntityTypeAttribute( tet, teaA, true, false );
         TrackedEntityTypeAttribute tetaB = new TrackedEntityTypeAttribute( tet, teaB, true, false );
         tet.getTrackedEntityTypeAttributes().add( tetaA );
         tet.getTrackedEntityTypeAttributes().add( tetaB );
-        currentSession.save( tet );
+        manager.save( tet );
         OrganisationUnit ou = createOrganisationUnit( 'a' );
+        manager.save( ou );
         TrackedEntityInstance tei = createTrackedEntityInstance( 'a', ou, teaA );
         tei.setTrackedEntityType( tet );
         TrackedEntityAttributeValue teavB = createTrackedEntityAttributeValue( 'b', tei, teaB );
-        tei.getTrackedEntityAttributeValues().add( teavB );
         TrackedEntityAttributeValue teavA = createTrackedEntityAttributeValue( 'a', tei, teaA );
-        tei.getTrackedEntityAttributeValues().add( teavA );
-        currentSession.save( ou );
-        currentSession.save( tei );
-        currentSession.save( teavA );
-        currentSession.save( teavB );
+        manager.save( tei );
+        trackedEntityAttributeValueService.addTrackedEntityAttributeValue( teavA );
+        trackedEntityAttributeValueService.addTrackedEntityAttributeValue( teavB );
+        tei.getTrackedEntityAttributeValues().addAll( List.of( teavA, teavB ) );
+        manager.update( tei );
     }
 
     @Override
     public void setUpTest()
     {
         userService = _userService;
-        currentSession = sessionFactory.getCurrentSession();
         User user = createUserWithAuth( "userUID0001" );
-        currentSession.save( user );
-        subject = new JacksonTrackedEntityInstanceService( teiService, trackedEntityAttributeService,
```

Comment:
is the subject never used?

Reply:
It is used but we can autowire it instead of manually created and manually inject all the dependencies